doc/developer/platform/ux.md

```diff
@@ -1,20 +1,945 @@
+document match the code, and started by replacing all use of "region" in this
+doc with "environment". In doing so I discovered that replacement made the
+user-facing concept substantially less clear, even though it has made the code
+substantially more clear. A "region" is a concept that is immediately understood
+by a user, whereas an "environment" requires explanation. So I think "region" is
+the term we should use in all user-facing material.
+
+Additionally, I think we should initially constrain users to a single
+environment per cloud-provider region, for the simple reason that it is easier
+to explain. Between databases and schemas, we already have several
+user-definable layers in the hierarchy, and I think adding another will cause
+confusion. RBAC with databases, schemas, and clusters meet the original design
+goal for environments, which was to allow customers to silo their staging and
+production infrastructure if they so desired.
+
+If we do discover compelling use cases for allowing multiple environments bound
+to the same region, it's easy to allow that later, whereas the reverse would be
+a painful breaking change.
+
+### Why SQL?
+
+The idea of the region API allowing only SQL queries has been met with some
+skepticism. Here are the arguments:
+
+* There is only one stable API. We don't need to worry about a separate HTTP API
+  for the web UI that needs its own versioning and backwards compatibility
+  guarantees. We just keep the SQL language stable and the system catalog
+  stable, which we already have to do to avoid breaking our customers. The web
+  UI then gets to piggyback on this stability for free.
+
+* Unified access control. The same SQL access control rules that regulate what
+  actions a user can take from a SQL shell regulate what actions they can take
+  from the web UI.
+
+* We get extremely valuable experience using Materialize to power a web UI,
+  which is exactly the use case we're selling.
+
+  The APIs we develop to support our own web UI, like tailing cluster
+  provisioning updates over WebSockets, are APIs that will also be useful to our
+  customers.
+
+I'd like to better understand the arguments against this approach. Here is what
+I recall:
+
+  * SQL won't integrate well with monitoring tools.
+
+    Counterpoint: we're going to need to get good at monitoring SQL queries in
+    order to support our customers.
+
+  * SQL won't integrate well with TypeScript.
+
+    Counterpoint: Nikhil will personally guarantee that we can generate
+    TypeScript bindings for the SQL API.
+
+#### SQL access control
+
+The approach of integrating Materialize's custom object types into the SQL
+access control system is modeled after [Snowflake's Access Control
+Privileges][snowflake-acl].
+
+I like this approach for several reasons:
+
+  * We don't need to invent a new access control model.
+
+  * We'll likely need to build support for `GRANT` and `REVOKE` for
```

Comment:
Will these need to be run for a specific user (email address)? How does the first user get access to be able to grant other users access? Should there be a way to generalize things so that you can grant access to certain frontegg roles?

Reply:
@mjibson and I talked about this offline, and I think we have a plan. The tl;dr is that "account owners" are treated as superusers in every region, are auto-created when they try to connect to a region, and can set up permissions for "account members" as they like.